src/test/kotlin/cz/petrbalat/spring/mvc/test/dsl/controller/DslControllerTestV2.kt

```diff
@@ -0,0 +1,156 @@
+package cz.petrbalat.spring.mvc.test.dsl.controller
+
+import cz.petrbalat.spring.mvc.test.dsl.*
+import org.hamcrest.CoreMatchers.`is`
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest
+import org.springframework.http.HttpMethod
+import org.springframework.http.HttpMethod.*
+import org.springframework.http.MediaType
+import org.springframework.test.context.ActiveProfiles
+import org.springframework.test.context.junit.jupiter.SpringExtension
+import org.springframework.test.web.servlet.*
+import org.springframework.test.web.servlet.request.MockHttpServletRequestBuilder
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders
+import org.springframework.test.web.servlet.result.MockMvcResultHandlers.*
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers
+import org.springframework.test.web.servlet.result.StatusResultMatchers
+import javax.servlet.http.Cookie
+
+@ExtendWith(SpringExtension::class)
+@WebMvcTest(HelloController::class)
+@ActiveProfiles("test")
+class DslControllerTestV2Test {
+
+    @Autowired
+    lateinit var mockMvc: MockMvc
+
+    @Test
+    fun helloGet() {
+        mockMvc.request { PUT("/hello") {
+                builder {
+                    contentType(MediaType.APPLICATION_JSON)
+                    content("""{"surname": "Jack"}""")
+                    cookie(Cookie("cookieName", "Extra Things"))
+                }
+                printRequestAndResponse()
+
+                expect {
+                    status { isBadRequest }
+                    "$.surname" jsonPath "Jack"
+                }
+
+                actions {
+                    andDo(print())
+                    expectContent { contentTypeCompatibleWith(MediaType.APPLICATION_JSON) }
+                }
+            }
+        }
+
+
```

Comment:
The example from lines 53-66 is my preferred new syntax.  The HttpMethod and url are required arguments. Then `builder` and `actions` expose the raw MockHttpServletRequestBuilder and ResultActions. Then we also add in a targeted `expect` and matching builder so we can clarify the expectations better (like the simplified jsonPath example on line `42`).

Reply:
I'll delete the `mockMvc.request { PUT("/hello") {` option. It was an experiment. But I don't like it because:

- adds an extra level of braces
- isn't obvious how to use, since the compiler doesn't prompt the user for the missing parts